db/migrate/20160509181733_create_security_logs.rb

```diff
@@ -0,0 +1,19 @@
+class CreateSecurityLogs < ActiveRecord::Migration
+  def change
+    create_table :security_logs do |t|
+      t.references :user
+      t.references :application
+      t.string :remote_ip, null: false
+      t.integer :event_type, null: false
+      t.text :event_data, null: false, default: '{}'
+
+      t.timestamps null: false
```

Comment:
Just have `created_at` by itself?  If so, block updates with a `before { false }` in `SecurityLog`?